site/content/how-to/traffic-management/snippets.md

```diff
@@ -0,0 +1,390 @@
+    rules with verbs `list` and `watch`, and add `snippetsfilters/status` to the RBAC rules with verb `update`. See this [example manifest](https://raw.githubusercontent.com/nginxinc/nginx-gateway-fabric/main/deploy/snippets-filters/deploy.yaml) for clarification.
+
+- Save the public IP address and port of NGINX Gateway Fabric into shell variables:
+
+   ```text
+  GW_IP=XXX.YYY.ZZZ.III
+  GW_PORT=<port number>
+  ```
+
+  {{< note >}} In a production environment, you should have a DNS record for the external IP address that is exposed, and it should refer to the hostname that the gateway will forward for. {{< /note >}}
+
+- Create the coffee and tea example applications:
+
+  ```yaml
+  kubectl apply -f https://raw.githubusercontent.com/nginxinc/nginx-gateway-fabric/v1.4.0/examples/snippets-filter/app.yaml
+  ```
+
+- Create a Gateway:
+
+  ```yaml
+  kubectl apply -f https://raw.githubusercontent.com/nginxinc/nginx-gateway-fabric/v1.4.0/examples/snippets-filter/gateway.yaml
+   ```
+
+- Create HTTPRoutes for the coffee and tea applications:
+
+  ```yaml
+  kubectl apply -f https://raw.githubusercontent.com/nginxinc/nginx-gateway-fabric/v1.4.0/examples/snippets-filter/httproutes.yaml
+   ```
+
+- Test the configuration:
+
+  You can send traffic to the coffee and tea applications using the external IP address and port for NGINX Gateway Fabric.
+
+  Send a request to coffee:
+
+  ```shell
+  curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/coffee
+  ```
+
+  This request should receive a response from the coffee Pod:
+
+  ```text
+  Server address: 10.244.0.7:8080
+  Server name: coffee-76c7c85bbd-cf8nz
+  ```
+
+  Send a request to tea:
+
+  ```shell
+  curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/tea
+   ```
+
+  This request should receive a response from the tea Pod:
+
+  ```text
+  Server address: 10.244.0.6:8080
+  Server name: tea-76c7c85bbd-cf8nz
+  ```
+
+  Before we enable rate limiting, try sending multiple requests to coffee:
+
+  ```shell
+  for i in `seq 1 10`; do curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/coffee; done
+  ```
+
+  You should see all successful responses in quick succession as we configured any rate limiting rules yet.
+
+---
+
+## Configure rate limiting to the coffee application
+
+Configure rate limiting to the coffee application by adding the following `SnippetsFilter`:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.nginx.org/v1alpha1
+kind: SnippetsFilter
+metadata:
+  name: coffee-rate-limiting-sf
+spec:
+  snippets:
+    - context: http
+      value: limit_req_zone \$binary_remote_addr zone=coffeezone:10m rate=1r/s;
+    - context: http.server.location
+      value: limit_req zone=coffeezone burst=3 nodelay;
+EOF
+```
+
+The Snippet uses the NGINX `limit_req_module` to configure rate limiting for this HTTPRoute and the
```

Comment:
This is where the Snippet term gets confusing for me. This is a SnippetsFilter with two snippets defined. Can we make that clear in the description?

Reply:
I combined this with the below 

>I think it would be beneficial to describe how the snippets in the referenced SnippetsFilter are injected into the NGINX config.

Let me know of your thoughts on the changes.

Reply:
Resolving this in favor of newer comment